packages/core/src/__tests__/acceptance/extension-point.acceptance.ts

```diff
@@ -0,0 +1,131 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/core
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  BINDING_METADATA_KEY,
+  Context,
+  Getter,
+  MetadataInspector,
+} from '@loopback/context';
+import {expect} from '@loopback/testlab';
+import {addExtension, extensionPoint, extensions} from '../..';
+
+describe('extension point', () => {
+  describe('@extensionPoint', () => {
+    it('specifies name of the extension point', () => {
+      @extensionPoint('greeters')
+      class GreetingService {
+        @extensions()
+        public greeters: Getter<Greeter[]>;
+      }
+
+      expect(
+        MetadataInspector.getClassMetadata(
+          BINDING_METADATA_KEY,
+          GreetingService,
+        ),
+      ).to.eql({tags: {name: 'greeters'}});
+    });
+  });
+
+  describe('@extensions', () => {
+    let ctx: Context;
+
+    beforeEach(givenContext);
+
+    it('injects a getter function of extensions', async () => {
+      @extensionPoint('greeters')
+      class GreetingService {
+        @extensions()
+        public greeters: Getter<Greeter[]>;
+      }
+
+      ctx.bind('greeter-service').toClass(GreetingService);
+      registerGreeters('greeters');
+      const greeterService = await ctx.get<GreetingService>('greeter-service');
+      const greeters = await greeterService.greeters();
+      const languages = greeters.map(greeter => greeter.language);
+      expect(languages).to.containEql('en');
+      expect(languages).to.containEql('zh');
+    });
+
+    it('injects extensions based on the name tag of the extension point binding', async () => {
+      class GreetingService {
+        @extensions()
+        public greeters: Getter<Greeter[]>;
+      }
+      ctx
+        .bind('greeter-service')
+        .toClass(GreetingService)
+        .tag({name: 'greeters'}); // Tag the extension point with a name
```

Comment:
I find this design where the extension point is specified as binding name, as suboptimal. What if the GreetingService wanted to leverage multiple extension points?

```ts
class GreetingService {
  @extensions('greeters')
  public greeters(): Getter<Greeter[]>;

  @extensions('loggers')
  public loggers(): Getter<Logger[]>;
}
```

Also the tag name `name` is easy to misinterpret, there is no indication that `name` is a name of the extension point, as opposed to e.g. name of the class, name of the binding (the key), etc.

Reply:
In fact, the tag `name` is already defined as "artifact name":

https://github.com/strongloop/loopback-next/blob/c774ed1e7b0168029ca8ec63de5867b475b26e0f/packages/context/src/keys.ts#L18-L21

Is the concept of "artifact name" the same as "extension point name"? They look like two different things to me.

Reply:
I now use `extensionPoint` tag for the name of an extension point. It falls back to `name` tag or the class name.